tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/UpToDate/BuildUpToDateCheckTests.cs

```diff
@@ -1336,14 +1336,49 @@ static int HashItems(params (string itemType, string[] paths)[] items)
             }
         }
 
+        [Fact]
+        public async Task IsUpToDateAsync_True_InputNewerThatBuiltOutput_TargetFrameworkDoesNotMatchBuild()
+        {
+            var projectSnapshot = new Dictionary<string, IProjectRuleSnapshotModel>
+            {
+                [UpToDateCheckBuilt.SchemaName] = SimpleItems("BuiltOutputPath1")
+            };
+
+            var sourceSnapshot = new Dictionary<string, IProjectRuleSnapshotModel>
+            {
+                [Compile.SchemaName] = SimpleItems("ItemPath1")
+            };
+
+            var itemChangedTime = DateTime.UtcNow.AddMinutes(-4);
+            var outputTime = DateTime.UtcNow.AddMinutes(-3);
+            var inputTime = DateTime.UtcNow.AddMinutes(-2);
+            var lastCheckTime = DateTime.UtcNow.AddMinutes(-1);
```

Comment:
The up-to-date check currently updates this value each time it's called (on the subscription object IIRC).

We currently track that value across all implicitly active configurations. I think you'll need to change the code to track it per configuration to handle all cases correctly. Take a look at the tests that play with this value to see the kind of trap if helps avoid, and I think you'll be able to construct a test that will fail unless this is handled per configuration.

Reply:
The last check time is currently modelled here: https://github.com/dotnet/project-system/blob/c8137374c0f7e6458397627b530ddb14099cb6b9/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.Subscription.cs#L49-L61

Reply:
I see. It sounds like a possible scenario is:
1. The user explicitly builds a project that multi-targets frameworks A and B.
2. During the build a file that is only included in a build of A is updated before the output for A is produced.
   - At this point `lastCheck` < `inputs(A)` < `outputs(A)`. If we were to check *all* configurations now, A would be (correctly) seen as out-of-date.
3. The user then runs the output of framework B (with, say, Ctrl+F5). The up-to-date check skips checking A (since we're not building it) and reports B is up-to-date because `inputs(B)` < `lastCheck` < `outputs(B)`. We update `lastCheck`.
4. The user explicitly builds again. B is, of course, up-to-date. A is *not* up-to-date, but we report that it is because `inputs(A)` < `outputs(A)` < `lastCheck`.
   - We then erroneously skip building A because we thought it was up-to-date, even though it wasn't.

Does that sound right?

Reply:
@drewnoakes Alright, I've updated the code to address this concern.

Reply:
Apologies, I missed your comment. Yes this looks like a valid scenario, and the unit test looks good.